sw/device/lib/base/testing/BUILD

```diff
@@ -0,0 +1,55 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"]) 
+
+cc_library(
+    name = "testing",
+    defines = ["MOCK_MMIO=1",
+               "MOCK_CSR=1",
+               ],
+    hdrs = [
+        "mock_mmio.h",
+        "mock_mmio_test_utils.h",
+        ],
+    srcs = [
+        "mock_mmio.cc",
+        "//sw/device/lib/base:mmio_src",
+        ],
+    deps = [
+        "//sw/device/lib/base",
+        "@googletest//:gtest",
+        ],
+    )
```

Comment:
Similar to the `base` library, the `testing` library can include both the mock_mmio and mock_csr sources and provide a single library target which supplies both of these resources.

Reply:
Should this gate this PR?

How do we want to link those directories in the source-tree? should the combined target be in //sw/device/lib/base/testing or in //sw/device/silicon_creator/lib/base ?

are there tests that rely on the mock_csr I can use to validate my work ?

Reply:
nvm, I hadn't realized mock_csr got moved.